actions/graybox/process-docx-worker.js

```diff
@@ -110,31 +118,50 @@ async function processFiles({
     const processDocxErrors = [];
 
     // iterate through preview statuses, generate docx files and create promote & copy batches
-    Object.keys(previewStatuses).forEach(async (batchName) => {
+    const batchNames = Object.keys(previewStatuses).flat();
+    const allProcessingPromises = batchNames.map(async (batchName, index, array) => {
         const batchPreviewStatuses = previewStatuses[batchName];
 
         // Check if Step 2 finished, do the Step 3, if the batch status is 'initial_preview_done' then process the batch
         if (batchStatusJson[batchName] === 'initial_preview_done') {
-            const allPreviewPromises = batchPreviewStatuses.map(async (status) => {
+            for (let prevIndex = 0; prevIndex < batchPreviewStatuses.length; prevIndex += 1) {
+                const status = batchPreviewStatuses[prevIndex];
                 if (status.success && status.mdPath) { // If the file is successfully initial previewed and has a mdPath then process the file
+                    // eslint-disable-next-line no-await-in-loop
                     const response = await sharepoint.fetchWithRetry(`${status.mdPath}`, options);
-                    const content = await response.text();
+                    // eslint-disable-next-line no-await-in-loop
+                    let content = await response.text();
                     let docx;
 
+                    // write the regex to match the image url where it will have [image then something then ] then https then something then media_
+                    // eslint-disable-next-line no-useless-escape
+                    const imageRegex = /\[image.*\]\: https.*media_[0-9a-f].*\#width=*\&*/g;
+                    const imageRegexMatch = content.match(imageRegex);
+
+                    // If the file has image links with -graybox, remove the -graybox from the image links
```

Comment:
@arshadparwaiz Looks like this code is processing the embedded images. 
Embedded images should not be processed. It should be ignored from the graybox check as those URLs don't exist in docx file. Those get automatically generated by Franklin in MD and will be specific to the project where it is previewed from when the docx file gets previewed.